### DISCUSSION TITLE
검색 리펙토링

### DISCUSSION THREAD

**Discussion by 14KGun** [sparcs-kaist/taxi-front#219] [Ideas]
## 검색 결과 뒤로가기 이슈
  알파테스트 버그 리포트에서도 언급되었듯이, 검색 결과에서 뒤로가기를 눌렀을때 검색을 아예 벗어나버리는 이슈가 있습니다.
  검색 조건들을 `url`에서 관리하는 것이 어떠할 듯 합니다

## 검색 결과에서 페이징이나 인피니티 스크롤이 필요할까??
  저번 회의에서 모든 검색 결과를 back에서 부터 받아오고 front에서 정렬을 처리하자는 합의를 이루었습니다.
  그러면 검색 결과에서 인피니티 스크롤이나 페이징이 필요할까요??

## "모든 방 검색하기" 네이밍 이슈
  모든 방 검색하기를 눌러도 24시간 이내의 방이 검색되는 것으로 알고 있습니다. 사용자들에게 혼란을 줄 수 있을 것 같습니다.

## 검색 결과 정렬 옵션들
  - 검색 옵션 : 인원이 만원인 방 검색 결과에서 제거
  - 정렬 옵션
    - 출발 시간이 빠른 순
    - 이름 사전 순
    - 최대 인원 작은 순
    - 최대 인원 큰 순
   
  - 검색 결과 옵션들에 대해서 이 정도 떠오르는데  추가 옵션이나, 제거해야할 옵션, 의견 있으십니까??

@GrasshopperBears

**Comment by wanwanss** [comment]
- 모든 방 검색하기의 결과가 24시간 이내의 방이라면, 이름을 '하루 안에 출발하는 방' 또는 '24시간 내 출발' (좋은 네이밍이 생각나지 않네요 ㅎㅎ,,) 이런 식으로 바꾸면 좋을 것 같습니다. 
- 검색 결과 정렬은 개인적으로 방의 이름 사전 순은 필요하지 않다고 생각합니다. 
- 꽉찬 방은 제거해서 보는 필터 기능이 있으면 좋을 것 같습니다!

**Comment by GrasshopperBears** [comment]
- 검색조건을 url에서 관리하는 것은 좋을 것 같습니다!
- 페이징
  - 결과를 한 번에 가져오면 굳이 페이징을 하지 않아도 될 것 같습니다. `content-visibility`와 같은 속성만 잘 적용하면 될 것 같습니다.
- 24시간 이내의 방을 검색하는 것은 괜찮아보입니다. 아무 제약조건 없이 전체를 검색하면 서버에 부하가 생길 것 같기도 하고 굳이 사용자가 아무 필터도 없이 모든 결과를 보고 싶어 할지는 잘 모르겠네요.
  - 다만 검색 버튼의 텍스트를 `빠른 검색`과 같이 바꾸면 더 좋을 것 같습니다.
- 출발 시간, 이름, 최대 인원 모두 natural, reverse order로 정렬할 수 있게 하면 어떨까요?

**Comment by imYourChoi** [comment]
- 검색 결과가 길어질 때 페이징을 하지 않는 것은 ux적으로 옳지 않은 것 같습니다. 검색 결과를 가져와서 global state로 저장 후 slicing만 해서 방들을보여주는게 나을것 같습니다.
- 모든 방 검색하기의 원래 기획 의도는 모든 방을 볼 수 있게 했던 게 맞습니다. 모든 방 검색하기를 눌러도 24시간 이내의 방이 검색되는 것은 오류인 것 같습니다. 24시간 내의 방을 보여주는게 2개의 날짜를 동시에 보여줄수 있기 때문에 하나의 날짜를 고르는 날짜 검색 옵션과 혼란이 있을 수 있습니다. 모든 방 보기가 아니라 디폴트를 "오늘 출발하는 방"으로 바꾸는 게 좋을것 같습니다. (하루의 기준이 새벽 5시까지였던 것 같아 24시간과 크게 다르지 않지만 조금 더 명확할 것 같습니다)
- 정렬 옵션은 출발 시간 순, 이름 오름차순/내림차순까지는 좋을 것 같고 탑슨 인원은 조금 생각해봐야 될 것 같습니다. 저희가 검색 옵션에 인원이 있기 때문에 차라리 남은 인원 순으로 하는게 더 자연스러울 것 같습니다. 만원인 방을 보여주는 건 좋은 것 같습니다

**Comment by 14KGun** [comment]
## 검색 결과 뒤로가기 이슈
  - `url`에서 관리

## 검색 결과에서 페이징이나 인피니티 스크롤이 필요할까??
  - 페이징

## "모든 방 검색하기" 네이밍 이슈
  모든 방 검색하기를 눌러도 24시간 이내의 방이 검색되는 것으로 알고 있습니다. 사용자들에게 혼란을 줄 수 있을 것 같습니다.
  - ~~@ptptsw, @0ev : 24시간 내 출발~~
    - 이 버튼이 의미가 있는지(??)
  - ~~@14KGun : 24시간 내 출발 방 검색하기~~
  - @GrasshopperBears : ~~빠른 검색, 간편 검색,~~ **빠른 출발 검색**
  - ~~@imYourChoi : 도움말, 설명서(툴팁) 추가~~

## 인원이 만원인 방 검색 결과에서 제거
 - @ptptsw : 검색 결과 나온 후 따로 체크 박스 만들기
 - ~~@imYourChoi : 검색 옵션에 추가~~
   - @GrasshopperBears : 검색 옵션(조건)에 추가 된다면, API호출에 여러번 될 수 있음

## 검색 결과 정렬 옵션들
  - 정렬 옵션
    - 출발 시간이 빠른 순
    - ~~이름 사전 순~~
    - ~~최대 인원 작은 순~~
    - ~~최대 인원 큰 순~~
    - 남은 인원 많은 순
    - 남은 인원 적은 순
